udp.go

```diff
@@ -0,0 +1,201 @@
+package tunnel
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/nknorg/nkn-sdk-go"
+	ts "github.com/nknorg/nkn-tuna-session"
+	"github.com/nknorg/tuna"
+)
+
+var (
+	ErrUDPWithTuna = errors.New("UDP is only supported in tuna mode")
+)
+
+// Generic interface for UDP conneciton, compatabile to net.UDPConn, nkn-tuna-session.
+type udpConn interface {
+	ReadFrom(b []byte) (n int, addr net.Addr, err error)
+	WriteTo(b []byte, addr net.Addr) (n int, err error)
+	Close() error
+}
+
+// UdpConn with updated (read or read) time.
+// Because UDP is connection-less, we don't know the peer endpoint is alive only when we get data from it.
+// So we purge idle UDP which has no read or write data in configured duration.
+type udpConnWithTime struct {
+	conn udpConn
+
+	sync.RWMutex
+	updTime time.Time // last updated time

```

Comment:
Suggest rename to `lastUpdateTime`, the current name looks like typo of udp

Reply:
Yes, renamed.